Office Addin - Module 5 - Style guide & QA refresh

andrewconnell opened the pull request:
This submission covers two primary updates:

- updated module to reflective guidance in the [Microsoft style guide](https://docs.microsoft.com/en-us/style-guide/welcome/)
- refreshed all content, labs, screenshots and demos to ensure they are current

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/README.md:
````diff
@@ -190,524 +128,574 @@ In this section, you will finish developing the Office Add-in using React and Ty
+    1. Add the following constant after the `import` statements and update the **{{REPLACE_WITH_ALPHAVANTAGE_APIKEY}}** to use your API key.
+
+        ```typescript
+        const ALPHAVANTAGE_APIKEY: string = '{{REPLACE_WITH_ALPHAVANTAGE_APIKEY}}';
````
might need to be updated based on @Lauragra feedback.

andrewconnell replied:
See above... if this has to change, I need guidance on what service to use to collect stock quotes.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/README.md:
```diff
@@ -190,524 +128,574 @@ In this section, you will finish developing the Office Add-in using React and Ty
+          this.setState({ waiting: true });
+          this.tableUtil.getColumnData('Symbol').then(
+            async (columnData: string[]) => {
+              // make sure the symbol was found in the Excel table
```
Ensure the symbol...

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/README.md:
```diff
@@ -190,524 +128,574 @@ In this section, you will finish developing the Office Add-in using React and Ty
+
+1. Update the **App** component to leverage the methods you added to the `ExcelTableUtil` class.
+    1. Locate and open the **src/components/App.tsx** file.
+    1. Locate and update the `refreshSymbol()` method in the `App` class to refresh specifying symbol in the Excel table
```
Locate and update the `refreshSymbol()` method in the `App` class to specify a symbol to refresh in the Excel table?

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/README.md:
```diff
@@ -190,524 +128,574 @@ In this section, you will finish developing the Office Add-in using React and Ty
+          this.setState({ waiting: true });
+          this.tableUtil.getColumnData('Symbol').then(
+            async (columnData: string[]) => {
+              // make sure the symbol was found in the Excel table
```
Ensure the symbol...

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/config/webpack.common.js:
```diff
@@ -1,124 +1,127 @@
-            }
-        ])
-    ]
+const webpack = require('webpack');
```
missing copyright and license header

andrewconnell replied:
? I just created a brand new project I don't see any license / copyright comments in any of the `webpack.*.js` generated files

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/config/webpack.dev.js:
```diff
@@ -1,36 +1,29 @@
-        historyApiFallback: true
-    }
-});
+const fs = require('fs');
```
missing copyright and license header

andrewconnell replied:
? I just created a brand new project I don't see any license / copyright comments in any of the `webpack.*.js` generated files

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/config/webpack.prod.js:
```diff
@@ -1,33 +1,32 @@
-        })
-    ]
-});
+const webpack = require('webpack');
```
Global comment for code files: missing copyright and license header

andrewconnell replied:
I'll add the comments to files that have comments in the project.

andrewconnell replied:
The latest Office generator only puts a license & copyright comment in the `index.html` file so that's the only one I listed it in, hence why it was removed.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/src/components/app.tsx:
```diff
@@ -1,177 +1,232 @@
+  // Gets a quote by calling into the stock service
+  getQuote = async (symbol: string) => {
+    return new Promise((resolve, reject) => {
+      const queryEndpoint = `https://www.alphavantage.co/query?function=BATCH_STOCK_QUOTES&symbols=${escape(
```
@Lauragra - another instance.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/src/utils/ExcelTableUtil.tsx:
```diff
@@ -0,0 +1,165 @@
+        });
+      }).catch(() => {
+        if (forceCreate) {
+          // Unable to find table...create it
```
see related comment above

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/src/utils/ExcelTableUtil.tsx:
```diff
@@ -0,0 +1,165 @@
+            const sheet = context.workbook.worksheets.getActiveWorksheet();
+            // Add the new row
+            tableRef.rows.add(null, [data]);
+            // Autofit columns and rows if supported by API
```
see related comment above

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
```diff
@@ -1,48 +1,34 @@
-In this demo, you will build an Office Add-in using Angular, a popular JavaScript framework.
+In this exercise, you will develop an Office Add-in using Angular and TypeScript. You will provision a new project using the Angular CLI and Office Yeoman generator, develop the add-in using Office.js, and test the add-in in Office Online.
 
 ## Running the project
```
can you move pre-requisites before running the project please?

andrewconnell replied:
Addressed across all three

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
```diff
@@ -72,39 +58,44 @@ In this section you will use the Angular CLI, the Office Yeoman generator, and N
+    ![Office Yeoman Generator](../../Images/YeomanAngular.png)
 
-1. When then Yeoman generator completes, change directories to the project folder (ex: **cd excel-portfolio**) and open the folder in your favorite code editor (you can use the command "**code .**" for [Visual Studio Code](https://code.visualstudio.com/)).
+1. When then Yeoman generator completes, open the project folder in a code editor (you can use the command `code .` for [Visual Studio Code](https://code.visualstudio.com/)).
```
When the...

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
```diff
@@ -122,80 +115,37 @@ In this section you will use the Angular CLI, the Office Yeoman generator, and N
+### Develop the Office Add-in
 
-1. Open **src/app.css** and replace the entire file with the contents show below.
+1. Open **src/styles.css** and replace the entire file with the contents show below.
```
shown below.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
`````diff
@@ -340,433 +289,450 @@ In this section, you will finish developing the Office Add-in using Angular and
 1. Next, open **src/app/app.component.ts** and update it as follows.
 
-    ````typescript
+    ```typescript
`````
global comment - missing license and copywrite information.

andrewconnell replied:
All code files generated will include the license & copyright information for generated files where the Office generator created the files. However, the latest version does not put these comments at the top of all files anymore.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
```diff
@@ -340,433 +289,450 @@ In this section, you will finish developing the Office Add-in using Angular and
-        }
+1. The **app.component.ts** file has a number of placeholder functions that you will complete to get the add-in functioning.
+    1. Locate & open the **src/app/appcomponent.ts** file.
+    1. Add the following constant after the `import` statements and update the **{{REPLACE_WITH_ALPHAVANTAGE_APIKEY}}** to use your API key.
```
@Lauragra similar concern

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
```diff
@@ -340,433 +289,450 @@ In this section, you will finish developing the Office Add-in using Angular and
-                    }, (createError) => {
-                        reject(createError);
-                    });
+                  // Unable to find table...create it
```
see above comment

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/README.md:
```diff
@@ -340,433 +289,450 @@ In this section, you will finish developing the Office Add-in using Angular and
+                  const sheet = context.workbook.worksheets.getActiveWorksheet();
+                  // Add the new row
+                  tableRef.rows.add(null, [data]);
+                  // Autofit columns and rows if supported by API
```
see above comment

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/src/karma.conf.js:
```diff
@@ -4,24 +4,22 @@
 module.exports = function (config) {
```
@Rick-Kirkham  please review

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Demos/02 Office Add-in using Angular/src/polyfills.ts:
```diff
@@ -37,21 +37,39 @@
 /** IE10 and IE11 requires the following for NgClass support on SVG elements */
```
@Rick-Kirkham  please review

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Lab.md:
````diff
@@ -1,51 +1,47 @@
     ```
 
-## Completed Exercises
+* A free API key from [Alpha Vantage](https://www.alphavantage.co): Registration is free and you will use the API key when creating stock quote requests.
````
@Lauragra  same concern.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Lab.md:
```diff
@@ -1,51 +1,47 @@
-# Office Add-ins: Build an Office Add-in using modern JavaScript tools and techniques - 300 Level
+# Build an Office Add-in using modern JavaScript tools and techniques
```
@andrewconnell is the content duplicated in this file?

andrewconnell replied:
The demos are excerpts of the labs... the updates requested in this review will be applied to the labs as well.

Reezaali commented on OfficeAddin/05 Using modern JavaScript/Lab.md:
````diff
@@ -1791,33 +1917,33 @@ In this exercise, you will develop an Office Add-in using Vue.js and TypeScript.
+    ```typescript
     getQuote(symbol:string) {
         return new Promise((resolve, reject) => {
             let url = `https://estx.azurewebsites.net/api/quote/${symbol}`;
````
remove the azure url?

andrewconnell commented:
@Reezaali:
> What's the role of the "lock" files?

These are added by NPM ... they ensure the next time **npm install** is run, it acquires the same package versions.

andrewconnell commented on OfficeAddin/05 Using modern JavaScript/Demos/01 Office Add-in using React/README.md:
````diff
@@ -190,524 +128,574 @@ In this section, you will finish developing the Office Add-in using React and Ty
+    ```typescript
+    getQuote = async (symbol: string) => {
+      return new Promise((resolve, reject) => {
+        const queryEndpoint = `https://www.alphavantage.co/query?function=BATCH_STOCK_QUOTES&symbols=${escape(symbol)}&interval=1min&apikey=${ALPHAVANTAGE_APIKEY}`;
````
Addressed this via email.

andrewconnell commented:
@Reezaali Addressed all feedback above in commit https://github.com/OfficeDev/TrainingContent/pull/503/commits/b09caa6d7c7d2e7b1c94871f5a959eabe87f9648. 

Because all three samples use the same utility class and process, I made sure the code comments you requested updates to were applied to all three. Your comments above only addressed one sample (the React one).

Furthermore, I noticed the Vue sample, lab & demo steps somehow got missed. I made the updates, but somehow they weren't part of the PR. I reapplied the changes to that project in commit https://github.com/OfficeDev/TrainingContent/pull/503/commits/0dccef71d0c04b859626aa342384923158cb2adc.
